pages/api/user/[id].ts

```diff
@@ -14,19 +15,41 @@ export default async function handle(
 ): Promise<void> {
     if (req.method == "GET") {
         // Obtain user id
-        const { id } = req.query;
+        const { userId } = req.query;
 
         // obtain user with provided userId
-        const user = await getUser(id as string);
+        const user = await getUser(userId as string);
 
         if (!user) {
-            ResponseUtil.returnNotFound(res, `User with id ${id} not found.`);
+            ResponseUtil.returnNotFound(
+                res,
+                `User with id ${userId} not found.`,
+            );
             return;
         }
         ResponseUtil.returnOK(res, user);
         return;
+    } else if (req.method == "PUT") {
+        const session = await getSession({ req });
+        // TODO: define custom session callback to contain user ID
+        const userId = session.id;
+        const updatedUserData = {
+            id: userId,
+            first_name: req.body.first_name,
+            last_name: req.body.last_name,
+            role: req.body.role,
+            role_data: req.body.role_data,
+        };
+        const newUser = await updateUser(updatedUserData);
```

Comment:
If you're updating user, I don't think you should name it `newUser`. I think `updatedUser` would make more sense.